src/mappings/entities/mapAsset.ts

```diff
@@ -0,0 +1,436 @@
+    link: doc.uri,
+  }));
+  await asset.save();
+};
+
+const handleRemoveDocuments = async (params: Record<string, any>) => {
+  const { ticker, ids } = params;
+  const asset = await getAsset(ticker);
+  asset.documents = asset.documents.filter(doc => !ids.includes(doc.id));
+  await asset.save();
+};
+
+const handleUpdateIdentifiers = async (params: Record<string, any>) => {
+  const { ticker, identifiers: newIdentifiers } = params;
+  const asset = await getAsset(ticker);
+  asset.identifiers = formatAssetIdentifiers(newIdentifiers);
+  await asset.save();
+};
+
+const handleMakeDivisible = async (params: Record<string, any>) => {
+  const { ticker } = params;
+  const asset = await getAsset(ticker);
+  asset.isDivisible = true;
+  await asset.save();
+};
+
+const handleIssue = async (params: Record<string, any>) => {
+  const { ticker, amount } = params;
+  const asset = await getAsset(ticker);
+  const formattedAmount = chainAmountToBigNumber(amount);
+  const assetOwner = await AssetHolder.get(`${asset.ownerDid}:${ticker}`);
+  assetOwner.amount = new BigNumber(assetOwner.amount).plus(formattedAmount).toString();
+  asset.totalSupply = new BigNumber(asset.totalSupply).plus(formattedAmount).toString();
+  await Promise.all([asset.save(), assetOwner.save()]);
+};
+
+const handleRedeem = async (params: Record<string, any>) => {
+  const { ticker, value: amount } = params;
+  const asset = await getAsset(ticker);
+  const formattedAmount = chainAmountToBigNumber(amount);
+  const assetOwner = await AssetHolder.get(`${asset.ownerDid}:${ticker}`);
+  assetOwner.amount = new BigNumber(assetOwner.amount).minus(formattedAmount).toString();
+  asset.totalSupply = new BigNumber(asset.totalSupply).minus(formattedAmount).toString();
+  await Promise.all([asset.save(), assetOwner.save()]);
+};
+
+const handleFreeze = async (params: Record<string, any>) => {
+  const { ticker } = params;
+  const asset = await getAsset(ticker);
+  asset.isFrozen = true;
+  await asset.save();
+};
+
+const handleUnfreeze = async (params: Record<string, any>) => {
+  const { ticker } = params;
+  const asset = await getAsset(ticker);
+  asset.isFrozen = false;
+  await asset.save();
+};
+
+const handleAcceptAssetOwnershipTransfer = async (params: Record<string, any>) => {
+  const { authId } = params;
+  const authorization = await getAuthorization(authId);
+  if (!authorization.ticker) return;
+  const asset = await getAsset(authorization.ticker);
+  asset.ownerDid = authorization.to;
+  await asset.save();
+  await AssetPendingOwnershipTransfer.remove(authId);
+};
+// #endregion
+
+// #region ModuleIdEnum.Identity
+const handleAddPendingOwnership = async (params: Record<string, any>, extrinsic: any) => {
+  const { target: targetData, authorizationData } = params;
+  const id = extrinsic.events[0].event.data[3].toString();
+  let ticker: string;
+  const from = extrinsic.events[0].event.data[0].toString();
+  const to = targetData.Identity.toString();
```

Comment:
An authorization can target an Identity or an Account, so this should be something like:

```
let to: string;
  if (!targetData.Identity.isEmpty) {
    to = targetData.Identity.toString();
  } else {
    to = targetData.Account.toString();
  }
  ```
  Preferably wrapped in a helper. It might be the case the Asset transfer only targets Identities, regardless I think it's good to be defensive and be able to handle both cases here since thats what the type permits.

Reply:
Ownership transfer and full agency is only done using the DID. Since we're now checking for the auth types specific to ownership transfers, this shouldn't be an issue. We're also only keeping track of DIDs right now. Handling accounts would add complexity since multiple accounts could be tied to the same DID. We can add it later if required.

Reply:
Makes sense, I was thinking we'd just use the address instead, but that doesn't make sense for the asset holder.

I guess the main thing was to check for the right types of auths.